internal/step/environment.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2021 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package step
+
+import (
+	"fmt"
+
+	"github.com/go-vela/types/constants"
+	"github.com/go-vela/types/library"
+	"github.com/go-vela/types/pipeline"
+)
+
+// Environment attempts to update the environment variables
+// for the container based off the library resources.
+func Environment(c *pipeline.Container, b *library.Build, r *library.Repo, s *library.Step) error {
+	// check if container or container environment are empty
+	if c == nil || c.Environment == nil {
+		return fmt.Errorf("empty container provided for environment")
+	}
+
+	// check if the build provided is empty
+	if b != nil {
+		// check if the channel exists in the environment
+		channel, ok := c.Environment["VELA_CHANNEL"]
+		if !ok {
+			// set default for channel
+			channel = constants.DefaultRoute
+		}
+
+		// check if the workspace exists in the environment
+		workspace, ok := c.Environment["VELA_WORKSPACE"]
+		if !ok {
+			// set default for workspace
+			//
+			// TODO: consider making this a constant
+			workspace = "/vela"
```

Comment:
Default is not this anymore, right?

Reply:
Or it's just a fallback?

Reply:
That will be addressed once https://github.com/go-vela/types/pull/135 is pulled into this repo 👍